doc/command_guidelines.md

```diff
@@ -0,0 +1,63 @@
+# Command Guidelines
+
+This document describes the command guidelines for 'az' and applies to both CLI command modules and extensions.
+
+Guidelines marked (*) only apply to command modules, not extensions.
+
+If in doubt, ask!
+
+## General Patterns
+
+- Be consistent with POSIX tools (support piping, work with grep, awk, jq, etc.)
+- Support tab completion for parameter names and values (e.g. resource names)
+- Commands must follow a "[noun] [noun] [verb]" pattern
+- For nouns that only support a single verb, the command should be named as a single hyphenated verb-noun pair
+- All commands, command group and arguments must have descriptions
+- You must provide command examples for non-trivial commands
+- All commands must support all output types (json, tsv, table)
+- Provide custom table outputs for commands that don't provide table output automatically
+- Commands must return an object or dictionary (do not string, Boolean, etc. types)
+- Use `stdout` and `stderr` appropriately.
+- Command output must go to stdout, everything else to stderr (log/status/errors).
```

Comment:
Since this line defines what "appropriately" is for stdout and stderr, does line 20 need to be said?

Reply:
removed that line.